File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>
#include <iostream>
int gcd(int a, int b)
{
    if(b == 0) return a;
    else return gcd(b, a % b);
}

using namespace std;

vector<int> solution(int n, int m) {
    vector<int> answer;
    
    answer.push_back(gcd(n, m));
    cout << gcd(n, m) << endl;
    answer.push_back(n * m / gcd(n, m));
    
    return answer;
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cs
using System;

public class Solution {
    public int GCD(int a, int b)
    {
        while(b != 0)
        {
            int tmp = b;
            b = a % b;
            a = tmp;
        }
        return a;
    }
    
    public int LCM(int a, int b)
    {
        return a * b / GCD(a, b);
    }
    
    public int[] solution(int n, int m) {
        int[] answer = new int[] {};
        
        Array.Resize(ref answer, 2);
        answer[0] = GCD(n, m);
        answer[1] = LCM(n, m);
        
        return answer;
    }
}